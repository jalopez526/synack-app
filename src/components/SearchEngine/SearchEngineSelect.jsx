import React from "react";
import Select from "../../Components/Select";
import Option from "../../Components/Option";

const SearchEngineSelect = ({ onChange }) => {
  return (
    <Select onChange={onChange} data-testid="select" className="control">
      <Option data-testid="select-option">Select a search engine</Option>
      <Option data-testid="select-option" value="google">
        Google
      </Option>
      <Option data-testid="select-option" value="bing">
        Bing
      </Option>
      <Option data-testid="select-option" value="both">
        Google / Bing
      </Option>
    </Select>
  );
};

export default SearchEngineSelect;
