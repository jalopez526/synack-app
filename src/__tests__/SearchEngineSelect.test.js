import { render, fireEvent, waitFor } from "@testing-library/react";
import SearchEngineSelect from "../Components/SearchEngine/SearchEngineSelect";
import renderer from 'react-test-renderer';

describe("Testing SearchSelect component", () => {
  const onChange = jest.fn();

  it('renders correctly', () => {
    const tree = renderer
      .create(<SearchEngineSelect onChange={onChange} value="google" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("render Select component and selecting option", async () => {
    const { getByTestId, getAllByTestId } = render(<SearchEngineSelect onChange={onChange} />);
    const element = getByTestId("select");
    fireEvent.change(element, { target: { value: "google" } });

    let options = getAllByTestId("select-option");
    await waitFor(() => {
      expect(element).toBeInTheDocument();
      expect(onChange).toHaveBeenCalled();
      expect(options[1].selected).toBeTruthy();
    })
  });
});
