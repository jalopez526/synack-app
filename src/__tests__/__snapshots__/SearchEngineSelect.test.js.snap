// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing SearchSelect component renders correctly 1`] = `
<select
  className="control"
  data-testid="select"
  onChange={[MockFunction]}
>
  <option
    data-testid="select-option"
  >
    Select a search engine
  </option>
  <option
    data-testid="select-option"
    value="google"
  >
    Google
  </option>
  <option
    data-testid="select-option"
    value="bing"
  >
    Bing
  </option>
  <option
    data-testid="select-option"
    value="both"
  >
    Google / Bing
  </option>
</select>
`;
